server/sonar-process/src/main/java/org/sonar/process/Lifecycle.java

```diff
@@ -101,4 +112,16 @@ public boolean equals(@Nullable Object o) {
   public int hashCode() {
     return state.hashCode();
   }
+
+  public interface LifecycleListener {
+    /**
+     * Called when a transition from state {@code from} to state {@code to} failed.
+     */
+    void failedTransition(State from, State to);
```

Comment:
The unique existing implementation does nothing in the method `failedTransition`. This code is useless. I suggest to remove it.